storage/src/journal/variable.rs

```diff
@@ -1,2223 +1,1443 @@
+    /// it will be updated to match the data journal.
     pub async fn init(context: E, cfg: Config<V::Cfg>) -> Result<Self, Error> {
-        // Iterate over blobs in partition
-        let mut blobs = BTreeMap::new();
-        let stored_blobs = match context.scan(&cfg.partition).await {
-            Ok(blobs) => blobs,
-            Err(RError::PartitionMissing(_)) => Vec::new(),
-            Err(err) => return Err(Error::Runtime(err)),
-        };
-        for name in stored_blobs {
-            let (blob, size) = context.open(&cfg.partition, &name).await?;
-            let hex_name = hex(&name);
-            let section = match name.try_into() {
-                Ok(section) => u64::from_be_bytes(section),
-                Err(_) => return Err(Error::InvalidBlobName(hex_name)),
-            };
-            debug!(section, blob = hex_name, size, "loaded section");
-            let blob = Append::new(blob, size, cfg.write_buffer, cfg.buffer_pool.clone()).await?;
-            blobs.insert(section, blob);
+        // Validate that partitions are different to prevent blob name collisions
+        if cfg.data_partition == cfg.offsets_partition {
+            return Err(Error::InvalidConfiguration(format!(
+                "partition and offsets_partition must be different: both are '{}'",
+                cfg.data_partition
+            )));
         }
 
-        // Initialize metrics
-        let tracked = Gauge::default();
-        let synced = Counter::default();
-        let pruned = Counter::default();
-        context.register("tracked", "Number of blobs", tracked.clone());
-        context.register("synced", "Number of syncs", synced.clone());
-        context.register("pruned", "Number of blobs pruned", pruned.clone());
-        tracked.set(blobs.len() as i64);
+        let items_per_section = cfg.items_per_section.get();
+
+        // Initialize underlying variable data journal
+        let mut data = multijournal::Journal::init(
+            context.clone(),
+            multijournal::Config {
+                partition: cfg.data_partition,
+                compression: cfg.compression,
+                codec_config: cfg.codec_config,
+                buffer_pool: cfg.buffer_pool.clone(),
+                write_buffer: cfg.write_buffer,
+            },
+        )
+        .await?;
+
+        // Initialize offsets journal
+        let mut offsets = fixed::Journal::init(
+            context,
+            fixed::Config {
+                partition: cfg.offsets_partition,
+                items_per_blob: cfg.items_per_section,
+                buffer_pool: cfg.buffer_pool,
+                write_buffer: cfg.write_buffer,
+            },
+        )
+        .await?;
+
+        // Validate and repair offsets journal to match data journal
+        let (oldest_retained_pos, size) =
+            Self::repair_journals(&mut data, &mut offsets, items_per_section).await?;
```

Comment:
We use the term "align_" instead of "repair_" elsewhere for this kind of thing. If you prefer repair (I have no preference), suggest updating the use of align_ for consistency.

Reply:
Changed "repair" to "align" except I kept "crash repair" because that sounds more natural than "crash alignment"